draco/tests/test_hard.py

```diff
@@ -296,20 +287,739 @@ def test_point_tick_bar_without_x_or_y():
+
+    :- violation(_).
+    """
+    )
+
+    assert not is_satisfiable(
+        p
+        + """
+    entity(mark,root,1).
+    attribute((mark,type),1,line).
+    entity(encoding,1,2).
+    attribute((encoding,channel),2,y).
+
+    :- violation(_).
+    """
+    )
+
+    assert not is_satisfiable(
+        p
+        + """
+    entity(mark,root,1).
+    attribute((mark,type),1,area).
+    entity(encoding,1,2).
+    attribute((encoding,channel),2,x).
+
+    :- violation(_).
+    """
+    )
+
+    assert not is_satisfiable(
+        p
+        + """
+    entity(mark,root,1).
+    attribute((mark,type),1,area).
+    entity(encoding,1,2).
+
+    :- violation(_).
+    """
+    )
+
+
+def test_line_area_with_discrete():
+    b = hard.blocks["line_area_with_discrete"]
+    assert isinstance(b, Block)
+    p = b.program + helpers.program
+
+    assert is_satisfiable(
+        p
+        + """
+    entity(mark,0,1).
+    entity(scale,0,2).
+    entity(scale,0,3).
+    attribute((mark,type),1,line).
+    attribute((scale,channel),2,x).
+    attribute((scale,type),2,linear).
+    attribute((scale,channel),3,y).
+    attribute((scale,type),3,linear).
+
+    :- violation(_).
+    """
+    )
+
+    assert is_satisfiable(
+        p
+        + """
+    entity(mark,0,1).
+    entity(scale,0,2).
+    entity(scale,0,3).
+    attribute((mark,type),1,line).
+    attribute((scale,channel),2,x).
+    attribute((scale,type),2,categorical).
+    attribute((scale,channel),3,y).
+    attribute((scale,type),3,linear).
+
+    :- violation(_).
+    """
+    )
+
+    assert not is_satisfiable(
+        p
+        + """
+    entity(mark,0,1).
+    entity(scale,0,2).
+    entity(scale,0,3).
+    attribute((mark,type),1,line).
+    attribute((scale,channel),2,x).
+    attribute((scale,type),2,categorical).
+    attribute((scale,channel),3,y).
+    attribute((scale,type),3,ordinal).
+
+    :- violation(_).
+    """
+    )
+
+
+def test_bar_tick_continuous_x_y():
+    b = hard.blocks["bar_tick_continuous_x_y"]
+    assert isinstance(b, Block)
+    p = b.program + helpers.program
+
+    assert is_satisfiable(
+        p
+        + """
+    entity(mark,0,1).
+    attribute((mark,type),1,tick).
+    entity(encoding,1,2).
+    attribute((encoding,channel),2,x).
+    entity(scale,0,4).
+    attribute((scale,channel),4,x).
+    attribute((scale,type),4,linear).
+
+    :- violation(_).
+    """
+    )
+
+    assert is_satisfiable(
+        p
+        + """
+    entity(mark,0,1).
     attribute((mark,type),1,tick).
+    entity(encoding,1,2).
+    attribute((encoding,channel),2,x).
+    entity(encoding,1,3).
+    attribute((encoding,channel),3,y).
+    entity(scale,0,4).
+    attribute((scale,channel),4,x).
+    attribute((scale,type),4,linear).
+    entity(scale,0,5).
+    attribute((scale,channel),5,y).
+    attribute((scale,type),5,categorical).
+
+    :- violation(_).
+    """
+    )
+
+    assert not is_satisfiable(
+        p
+        + """
+    entity(view,root,1).
```

Comment:
Uff, this becomes hard to read. Can we make a smaller example or use the nested representation?

Reply:
Same for some of the specs above. Use your judgment.